src/Compilers/CSharp/Portable/Parser/QuickScanner.cs

```diff
@@ -209,7 +209,7 @@ private SyntaxToken QuickScanSyntaxToken()
                 var flags = uc < charPropLength ? (CharFlags)s_charProperties[uc] : CharFlags.Complex;
 
                 state = (QuickScanState)s_stateTransitions[(int)state, (int)flags];
-                if (state == QuickScanState.Done || state == QuickScanState.Bad)
+                if (state >= QuickScanState.Done)
```

Comment:
This is one of the hottest conditions in the scanner.
I will add an assert and a comment.